### PR TITLE
Registration work

### PR DESCRIPTION
**Pull request by linusarnejohansson**

### CODE REVIEW

File: src/components/login/index.tsx [hunk 1]
```diff
@@ -19,8 +19,9 @@ export class UserwidgetsLogin {
 			user: event.detail.user,
 			password: event.detail.password,
 		})
-		if (this.resolve != undefined && !gracely.Error.is(response)) {
+		if (this.resolve != undefined && response) {
 			this.resolve(true)
+			state.options = { user: response.email, applicationId: response.audience }
```
Comment (EliasEriksson) [inline src/components/login/index.tsx:24]: this is done inside `this.state.me.login()` already. unnecessary to also do it here.

File: src/components/menu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@Component({
+	tag: "userwidgets-menu",
+	styleUrl: "style.css",
+	scoped: true,
+})
+export class UserwidgetsMenu {
+	@Prop({ mutable: true, reflect: true }) menuOpen = false
+
+	handleClick() {
+		console.log("handleClick")
```
Comment (EliasEriksson) [inline src/components/menu/index.tsx:12]: remove console.log

File: src/components/menu/style.css [hunk 3]
```diff
@@ -0,0 +1,23 @@
+.menu{
+	position: absolute;
+	top: 5rem;
+	right: auto ;
```
Comment (EliasEriksson) [inline src/components/menu/style.css:4]: this is unnecessary now when we use transform.

File: src/components/menu/style.css [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	left: 100%;
+	justify-content: center;
+	align-items: center;
+	padding: 0 1em;
+	height: 100%;
+	margin: 1rem;
+	transition: 0.3s transform ease-in-out;
+	color: black;
```
Comment (EliasEriksson) [inline src/components/menu/style.css:12]: is this color a leftover for something or intentional?

File: src/components/menu/style.css [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	padding: 0 1em;
+	height: 100%;
+	margin: 1rem;
+	transition: 0.3s transform ease-in-out;
+	color: black; 
+}
+:host[menu-open] .menu{
+	transform: translate(-100%,0)
+}
+.menu > * {
```
Comment (EliasEriksson) [inline src/components/menu/style.css:17]: double selector on `.menu`

File: src/components/menu/style.css [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+	padding: 1rem;
```
Comment (EliasEriksson) [inline src/components/menu/style.css:18]: this value overwrites the padding from the other `.menu` selector. take one or the other not both at the same time.

File: src/components/menu/style.css [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+}
+.background {
+	background-color: rgba(255, 255, 255, 0.55);
```
Comment (EliasEriksson) [inline src/components/menu/style.css:21]: can we use a smoothly color instead? if not, ignore this.

File: src/components/organizationPicker/index.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+	@Watch("key")
+	handelKey() {
+		this.key &&
+			(this.organizations = Object.keys(this.key.permissions)
+				.filter(([organizationId]) => organizationId != "*")
+				.map(organizationId => ({ name: organizationId, value: organizationId }))) &&
+			(state.options = { organizationId: this.organizations[0].value })
+		console.log("this.key", this.key)
+		console.log("this.organizations", this.organizations)
+		console.log("state.options", state.options)
```
Comment (EliasEriksson) [inline src/components/organizationPicker/index.tsx:23]: remove these console logs

File: src/components/organizationPicker/index.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+	}
+
+	componentWillLoad() {
+		state.me.listen("key", async key => (this.key = await key))
+		console.log("key ", this.key)
+		console.log("organization picker componentWillLoad")
```
Comment (EliasEriksson) [inline src/components/organizationPicker/index.tsx:29]: remove console logs

File: src/components/organizationPicker/index.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+	}
+
+	@Listen("menuClose")
+	handleMenuClose(event: CustomEvent<OptionType[]>) {
+		event.stopPropagation()
+		console.log("handleMenuClose state", state.options.organizationId)
```
Comment (EliasEriksson) [inline src/components/organizationPicker/index.tsx:35]: remove console log

File: src/components/organizationPicker/index.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+		state.options = { organizationId: event.detail[0].value }
+	}
+	render() {
+		return !this.organizations ? (
+			<div>
+				<p>You are not a member of any organization.</p>
```
Comment (EliasEriksson) [inline src/components/organizationPicker/index.tsx:41]: can we use something else than a p tag for this? if not ignore this.

**Review by EliasEriksson — CHANGES_REQUESTED** [review]
minor things

**Comment by EliasEriksson** [comment]
Currently the component only shows IDs. It should also fetch the organizations and use their names for display. Using IDs in both places was a starting point. Ask me when you get to this point,

File: src/components/menu/style.css (hunk 3 above)
Comment (linusarnejohansson) [reply src/components/menu/style.css:4]: position absolute is necessary so is keeping that rule. top and right appears to be unnecessary so will remove those

File: src/State/Users.ts [hunk 12]
```diff
@@ -13,13 +14,22 @@ export class Users {
...
 		)
 	}
 	set users(users: Promise<model.userwidgets.User[] | undefined>) {
 		this.#users = users
 	}
-	constructor(private readonly client: Client, options?: Options) {
-		options && (this.options = options)
+	#client: Client
+	#self: Users & Listenable<Users>
+	constructor(listenable: Users & Listenable<Users>, client: Client) {
```
Comment (EliasEriksson) [inline src/State/Users.ts:27]: make the constructor private

File: src/components/organizationPicker/index.tsx [hunk 13]
```diff
@@ -8,44 +8,39 @@ import { state } from "../../State"
 	styleUrl: "style.css",
 	scoped: true,
 })
+// now use the new state application to fetch names
```
Comment (EliasEriksson) [inline src/components/organizationPicker/index.tsx:11]: remove comment

File: src/components/organizationPicker/index.tsx [hunk 14]
```diff
@@ -8,44 +8,39 @@ import { state } from "../../State"
...
 export class UserwidgetsOrganizationPicker {
 	@State() key?: model.userwidgets.User.Key
 	@State() organizations?: { name: string; value: string }[]
-	@Watch("key")
+	@State() application?: model.userwidgets.Application
+	@Watch("application")
 	handelKey() {
```
Comment (EliasEriksson) [inline src/components/organizationPicker/index.tsx:17]: change name to `handleAppliicationChange`, or `applicationChanged` or similar

File: src/components/organizationPicker/index.tsx [hunk 15]
```diff
@@ -8,44 +8,39 @@ import { state } from "../../State"
...
 	componentWillLoad() {
 		state.me.listen("key", async key => (this.key = await key))
-		console.log("key ", this.key)
-		console.log("organization picker componentWillLoad")
+		state.application.listen("application", async application => (this.application = await application))
 	}
 
 	@Listen("menuClose")
 	handleMenuClose(event: CustomEvent<OptionType[]>) {
-		event.stopPropagation()
```
Comment (EliasEriksson) [inline src/components/organizationPicker/index.tsx]: bring this back. it should not propagate higher in the DOM.

**Review by EliasEriksson — CHANGES_REQUESTED** [review]

**Pull request merged**